### PR TITLE
Nullable annotation fixes

### PR DESCRIPTION
**Pull request by jaredpar**
Once runtime merges to a Roslyn toolset that has the fix for [issue 50782](https://github.com/dotnet/roslyn/issues/50782) these annotations will be necessary. Front loading the work here with this change.

### CODE REVIEW

**Comment by jaredpar** [comment]
Can see the need for these demonstrated with #82615 where I was experimenting with latest compiler from `main`.

**Review by RikkiGibson — APPROVED** [review]

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs [hunk 1]
```diff
@@ -494,7 +494,7 @@ private void RemoveStalePools()
             {
                 if (entry.Value.CleanCacheAndDisposeIfUnused())
                 {
-                    _pools.TryRemove(entry.Key, out HttpConnectionPool _);
+                    _pools.TryRemove(entry.Key, out HttpConnectionPool? _);
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs:497]:
```suggestion
                    _pools.TryRemove(entry.Key, out _);
```
?

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs [hunk 2]
```diff
@@ -891,7 +891,7 @@ public bool HasSameRules(TimeZoneInfo other)
             string? displayName,
             string? standardDisplayName)
         {
-            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, allocate: false, out string _);
+            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, allocate: false, out string? _);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs:894]:
```suggestion
            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, allocate: false, out _);
```

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs [hunk 3]
```diff
@@ -942,7 +942,7 @@ public bool HasSameRules(TimeZoneInfo other)
                 adjustmentRules = (AdjustmentRule[])adjustmentRules.Clone();
             }
 
-            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, allocate: false, out string _);
+            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, allocate: false, out string? _);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs:945]:
```suggestion
            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, allocate: false, out _);
```
?

File: src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSet.cs [hunk 4]
```diff
@@ -650,7 +650,7 @@ internal void SetContractProcessed(DataContract dataContract)
         public void ImportSchemaSet(XmlSchemaSet schemaSet, IEnumerable<XmlQualifiedName>? typeNames, bool importXmlDataType)
         {
             SchemaImporter importer = new SchemaImporter(schemaSet, typeNames, null, this, importXmlDataType);
-            importer.Import(out List<XmlQualifiedName> _);
+            importer.Import(out List<XmlQualifiedName>? _);
```
Comment (stephentoub) [inline src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSet.cs:653]:
```suggestion
            importer.Import(out _);
```
?

File: src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Parameters/Ecma/EcmaFatMethodParameter.cs [hunk 5]
```diff
@@ -37,7 +37,7 @@ internal EcmaFatMethodParameter(IRoMethodBase roMethodBase, int position, Type p
 
         protected sealed override IEnumerable<CustomAttributeData> GetTrueCustomAttributes() => Parameter.GetCustomAttributes().ToTrueCustomAttributes(GetEcmaModule());
 
-        public sealed override bool HasDefaultValue => TryGetRawDefaultValue(out object _);
+        public sealed override bool HasDefaultValue => TryGetRawDefaultValue(out object? _);
```
Comment (stephentoub) [inline src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Parameters/Ecma/EcmaFatMethodParameter.cs:40]:
```suggestion
        public sealed override bool HasDefaultValue => TryGetRawDefaultValue(out _);
```
?

**Review by stephentoub — APPROVED** [review]
Unless the type is necessary to disambiguate, can all of these just be discards without the type name at all?

**Comment by jaredpar** [comment]
> Unless the type is necessary to disambiguate, can all of these just be discards without the type name at all?

Sure thing. Was just trying to maintain existing style. happy to move to discards.

**Pull request merged**